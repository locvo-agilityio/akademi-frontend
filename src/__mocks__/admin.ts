export const MOCK_ADMIN = {
  id: 1,
  name: 'Nabila A',
  role: 'Admin',
  avatar: 'https://bit.ly/dan-abramov',
};
