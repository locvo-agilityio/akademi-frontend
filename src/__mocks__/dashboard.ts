import {
  CalendarIcon,
  FoodIcon,
  StudentIcon,
  TeacherIcon,
} from '@/components/icons';

export const MOCK_SUMMARY_DASHBOARD = [
  {
    title: 'Students',
    total: 932,
    bgIcon: 'primary',
    icon: StudentIcon,
  },
  {
    title: 'Teachers',
    total: 754,
    bgIcon: 'secondary',
    icon: TeacherIcon,
  },
  {
    title: 'Events',
    total: 40,
    bgIcon: 'yellow',
    icon: CalendarIcon,
  },
  {
    title: 'Foods',
    total: 32,
    bgIcon: 'darkBlue',
    icon: FoodIcon,
  },
];
