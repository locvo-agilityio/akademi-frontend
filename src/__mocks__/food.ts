export const MOCK_FOODS = [
  {
    name: 'Beef Steak with Fried Potato',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    image:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRCQPUcFz7CxVScB-zxL5pl2w6f4ntjw4mV-w&s',
  },
  {
    name: 'Beef Steak with Fried Potato New',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    image:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRCQPUcFz7CxVScB-zxL5pl2w6f4ntjw4mV-w&s',
  },
];
