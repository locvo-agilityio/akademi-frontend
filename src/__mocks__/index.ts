export * from './admin';
export * from './dashboard';
export * from './student';
export * from './pagination';
export * from './message';
export * from './food';
export * from './teacher';
export * from './schedule';
export * from './payment';
