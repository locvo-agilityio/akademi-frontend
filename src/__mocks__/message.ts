export const MOCK_MESSAGES = [
  {
    name: 'Samantha William',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
  {
    name: 'Tony Soap',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
  {
    name: 'Jordan Nico',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
  {
    name: 'Nadila Adja',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
];
