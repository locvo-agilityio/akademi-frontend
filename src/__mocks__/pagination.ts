export const MOCK_PAGINATION = {
  totalRecords: '100 items',
  pageSize: 10,
  currentButtons: [1, 2, 3],
  currentPage: 1,
};
