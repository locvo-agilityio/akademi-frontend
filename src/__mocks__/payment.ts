export const MOCK_PAYMENTS = [
  {
    id: 1,
    date: String(new Date()),
    amount: 100000,
    status: 'Pending',
  },
  {
    id: 2,
    date: String(new Date()),
    amount: 100000,
    status: 'Complete',
  },
  {
    id: 3,
    date: String(new Date()),
    amount: 100000,
    status: 'Canceled',
  },
  {
    id: 4,
    date: String(new Date()),
    amount: 100000,
    status: 'Complete',
  },
];
