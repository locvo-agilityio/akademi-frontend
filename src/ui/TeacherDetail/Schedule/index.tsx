import { memo } from 'react';
import isEqual from 'react-fast-compare';
import { Button, Flex, Text, VStack } from '@chakra-ui/react';

// Components
import { CardSchedule } from '@/components';

// Utils
import { formatDate } from '@/utils';

// Types
import { ISchedule } from '@/types';

interface IScheduleProps {
  schedules?: ISchedule[];
}

const ScheduleTeacher = ({ schedules = [] }: IScheduleProps) => (
  <VStack display={{ base: 'none', xl: 'flex' }} w={418} gap={6}>
    <Flex
      w="full"
      direction="column"
      borderRadius="2xl"
      bgColor="white"
      gap={2}
      p={8}
    >
      <Text as="h3" size="lg" fontWeight="bold" color="darkBlue">
        Schedule Details
      </Text>
      <Text as="p" color="whiteSmoke">
        {formatDate(String(new Date()))}
      </Text>
    </Flex>

    {schedules.map((item) => (
      <CardSchedule key={item.id} {...item} />
    ))}

    <Button
      w="full"
      mt={6}
      p={7}
      variant="tertiary"
      borderRadius="full"
      disabled
    >
      View More
    </Button>
  </VStack>
);

export default memo(ScheduleTeacher, isEqual);
