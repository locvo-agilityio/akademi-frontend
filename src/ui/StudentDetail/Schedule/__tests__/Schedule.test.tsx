import { render } from '@testing-library/react';

// Components
import Schedule from '..';

// Mocks
import { MOCK_STUDENT_SCHEDULES } from '@/__mocks__';

describe('Schedule', () => {
  it('should snapshot default value', () => {
    const { container } = render(<Schedule />);

    expect(container).toBeInTheDocument();
  });

  it('should snapshot successfully', () => {
    const { container } = render(
      <Schedule schedules={MOCK_STUDENT_SCHEDULES} />,
    );

    expect(container).toMatchSnapshot();
  });
});
