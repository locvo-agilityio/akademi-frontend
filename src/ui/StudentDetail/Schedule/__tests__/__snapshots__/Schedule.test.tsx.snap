// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule should snapshot successfully 1`] = `
<div>
  <div
    class="chakra-stack css-qtenzj"
  >
    <div
      class="css-1edhruq"
    >
      <h3
        class="chakra-text css-tbw3nr"
      >
        Schedule Details
      </h3>
      <p
        class="chakra-text css-cif1vy"
      >
        February 10, 2025
      </p>
    </div>
    <div
      class="chakra-card css-1qa4d5p"
    >
      <div
        class="chakra-card__header css-old1by"
      >
        <p
          class="chakra-text css-1nveigd"
        >
          Basic Algorithm
        </p>
        <span
          class="chakra-text css-bxh4bx"
        >
          Algorithm
        </span>
      </div>
      <div
        class="chakra-card__body css-19n62ks"
      >
        <div
          class="css-17dp5mp"
        >
          <div
            class="css-19p4nd6"
          >
            <svg
              class="chakra-icon css-ivbvq"
              fill="none"
              focusable="false"
              height="40"
              viewBox="0 0 40 40"
              width="40"
            >
              <path
                d="M30 8.75H27.5V7.5C27.5 7.16848 27.3683 6.85054 27.1339 6.61612C26.8995 6.3817 26.5815 6.25 26.25 6.25C25.9185 6.25 25.6005 6.3817 25.3661 6.61612C25.1317 6.85054 25 7.16848 25 7.5V8.75H21.25V7.5C21.25 7.16848 21.1183 6.85054 20.8839 6.61612C20.6495 6.3817 20.3315 6.25 20 6.25C19.6685 6.25 19.3505 6.3817 19.1161 6.61612C18.8817 6.85054 18.75 7.16848 18.75 7.5V8.75H15V7.5C15 7.16848 14.8683 6.85054 14.6339 6.61612C14.3995 6.3817 14.0815 6.25 13.75 6.25C13.4185 6.25 13.1005 6.3817 12.8661 6.61612C12.6317 6.85054 12.5 7.16848 12.5 7.5V8.75H10C9.00544 8.75 8.05161 9.14509 7.34835 9.84835C6.64509 10.5516 6.25 11.5054 6.25 12.5V30C6.25 30.9946 6.64509 31.9484 7.34835 32.6517C8.05161 33.3549 9.00544 33.75 10 33.75H30C30.9946 33.75 31.9484 33.3549 32.6517 32.6517C33.3549 31.9484 33.75 30.9946 33.75 30V12.5C33.75 11.5054 33.3549 10.5516 32.6517 9.84835C31.9484 9.14509 30.9946 8.75 30 8.75ZM8.75 12.5C8.75 12.1685 8.8817 11.8505 9.11612 11.6161C9.35054 11.3817 9.66848 11.25 10 11.25H12.5V12.5C12.5 12.8315 12.6317 13.1495 12.8661 13.3839C13.1005 13.6183 13.4185 13.75 13.75 13.75C14.0815 13.75 14.3995 13.6183 14.6339 13.3839C14.8683 13.1495 15 12.8315 15 12.5V11.25H18.75V12.5C18.75 12.8315 18.8817 13.1495 19.1161 13.3839C19.3505 13.6183 19.6685 13.75 20 13.75C20.3315 13.75 20.6495 13.6183 20.8839 13.3839C21.1183 13.1495 21.25 12.8315 21.25 12.5V11.25H25V12.5C25 12.8315 25.1317 13.1495 25.3661 13.3839C25.6005 13.6183 25.9185 13.75 26.25 13.75C26.5815 13.75 26.8995 13.6183 27.1339 13.3839C27.3683 13.1495 27.5 12.8315 27.5 12.5V11.25H30C30.3315 11.25 30.6495 11.3817 30.8839 11.6161C31.1183 11.8505 31.25 12.1685 31.25 12.5V16.25H8.75V12.5ZM31.25 30C31.25 30.3315 31.1183 30.6495 30.8839 30.8839C30.6495 31.1183 30.3315 31.25 30 31.25H10C9.66848 31.25 9.35054 31.1183 9.11612 30.8839C8.8817 30.6495 8.75 30.3315 8.75 30V18.75H31.25V30Z"
                fill="currentColor"
              />
            </svg>
            <span
              class="chakra-text css-0"
            >
              February 10, 2025
            </span>
          </div>
          <span
            class="css-19p4nd6"
          >
            <svg
              class="chakra-icon css-1fygt0h"
              fill="none"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M10 0C8.02219 0 6.08879 0.58649 4.4443 1.6853C2.79981 2.78412 1.51809 4.3459 0.761209 6.17317C0.00433284 8.00043 -0.193701 10.0111 0.192152 11.9509C0.578004 13.8907 1.53041 15.6725 2.92894 17.0711C4.32746 18.4696 6.10929 19.422 8.0491 19.8079C9.98891 20.1937 11.9996 19.9957 13.8268 19.2388C15.6541 18.4819 17.2159 17.2002 18.3147 15.5557C19.4135 13.9112 20 11.9778 20 10C20 8.68678 19.7413 7.38642 19.2388 6.17317C18.7363 4.95991 17.9997 3.85752 17.0711 2.92893C16.1425 2.00035 15.0401 1.26375 13.8268 0.761205C12.6136 0.258658 11.3132 0 10 0ZM10 18C8.41775 18 6.87104 17.5308 5.55544 16.6518C4.23985 15.7727 3.21447 14.5233 2.60897 13.0615C2.00347 11.5997 1.84504 9.99113 2.15372 8.43928C2.4624 6.88743 3.22433 5.46197 4.34315 4.34315C5.46197 3.22433 6.88743 2.4624 8.43928 2.15372C9.99113 1.84504 11.5997 2.00346 13.0615 2.60896C14.5233 3.21447 15.7727 4.23984 16.6518 5.55544C17.5308 6.87103 18 8.41775 18 10C18 12.1217 17.1572 14.1566 15.6569 15.6569C14.1566 17.1571 12.1217 18 10 18Z"
                fill="#FCC43E"
              />
              <path
                d="M13 9H11V5C11 4.73478 10.8946 4.48043 10.7071 4.29289C10.5196 4.10536 10.2652 4 10 4C9.73478 4 9.48043 4.10536 9.29289 4.29289C9.10536 4.48043 9 4.73478 9 5V10C9 10.2652 9.10536 10.5196 9.29289 10.7071C9.48043 10.8946 9.73478 11 10 11H13C13.2652 11 13.5196 10.8946 13.7071 10.7071C13.8946 10.5196 14 10.2652 14 10C14 9.73478 13.8946 9.48043 13.7071 9.29289C13.5196 9.10536 13.2652 9 13 9Z"
                fill="#FCC43E"
              />
            </svg>
            <span
              class="chakra-text css-0"
            >
              07.00 - 09.00 AM
            </span>
          </span>
        </div>
        <div
          class="chakra-stack css-tl3ftk"
        >
          <span
            class="chakra-avatar css-1ruhx4w"
          >
            <div
              aria-label="Algorithm"
              class="chakra-avatar__initials css-0"
              role="img"
            >
              A
            </div>
          </span>
        </div>
      </div>
    </div>
    <div
      class="chakra-card css-15dgxy4"
    >
      <div
        class="chakra-card__header css-old1by"
      >
        <p
          class="chakra-text css-1nveigd"
        >
          Basic Art
        </p>
        <span
          class="chakra-text css-bxh4bx"
        >
          Art
        </span>
      </div>
      <div
        class="chakra-card__body css-19n62ks"
      >
        <div
          class="css-17dp5mp"
        >
          <div
            class="css-19p4nd6"
          >
            <svg
              class="chakra-icon css-ivbvq"
              fill="none"
              focusable="false"
              height="40"
              viewBox="0 0 40 40"
              width="40"
            >
              <path
                d="M30 8.75H27.5V7.5C27.5 7.16848 27.3683 6.85054 27.1339 6.61612C26.8995 6.3817 26.5815 6.25 26.25 6.25C25.9185 6.25 25.6005 6.3817 25.3661 6.61612C25.1317 6.85054 25 7.16848 25 7.5V8.75H21.25V7.5C21.25 7.16848 21.1183 6.85054 20.8839 6.61612C20.6495 6.3817 20.3315 6.25 20 6.25C19.6685 6.25 19.3505 6.3817 19.1161 6.61612C18.8817 6.85054 18.75 7.16848 18.75 7.5V8.75H15V7.5C15 7.16848 14.8683 6.85054 14.6339 6.61612C14.3995 6.3817 14.0815 6.25 13.75 6.25C13.4185 6.25 13.1005 6.3817 12.8661 6.61612C12.6317 6.85054 12.5 7.16848 12.5 7.5V8.75H10C9.00544 8.75 8.05161 9.14509 7.34835 9.84835C6.64509 10.5516 6.25 11.5054 6.25 12.5V30C6.25 30.9946 6.64509 31.9484 7.34835 32.6517C8.05161 33.3549 9.00544 33.75 10 33.75H30C30.9946 33.75 31.9484 33.3549 32.6517 32.6517C33.3549 31.9484 33.75 30.9946 33.75 30V12.5C33.75 11.5054 33.3549 10.5516 32.6517 9.84835C31.9484 9.14509 30.9946 8.75 30 8.75ZM8.75 12.5C8.75 12.1685 8.8817 11.8505 9.11612 11.6161C9.35054 11.3817 9.66848 11.25 10 11.25H12.5V12.5C12.5 12.8315 12.6317 13.1495 12.8661 13.3839C13.1005 13.6183 13.4185 13.75 13.75 13.75C14.0815 13.75 14.3995 13.6183 14.6339 13.3839C14.8683 13.1495 15 12.8315 15 12.5V11.25H18.75V12.5C18.75 12.8315 18.8817 13.1495 19.1161 13.3839C19.3505 13.6183 19.6685 13.75 20 13.75C20.3315 13.75 20.6495 13.6183 20.8839 13.3839C21.1183 13.1495 21.25 12.8315 21.25 12.5V11.25H25V12.5C25 12.8315 25.1317 13.1495 25.3661 13.3839C25.6005 13.6183 25.9185 13.75 26.25 13.75C26.5815 13.75 26.8995 13.6183 27.1339 13.3839C27.3683 13.1495 27.5 12.8315 27.5 12.5V11.25H30C30.3315 11.25 30.6495 11.3817 30.8839 11.6161C31.1183 11.8505 31.25 12.1685 31.25 12.5V16.25H8.75V12.5ZM31.25 30C31.25 30.3315 31.1183 30.6495 30.8839 30.8839C30.6495 31.1183 30.3315 31.25 30 31.25H10C9.66848 31.25 9.35054 31.1183 9.11612 30.8839C8.8817 30.6495 8.75 30.3315 8.75 30V18.75H31.25V30Z"
                fill="currentColor"
              />
            </svg>
            <span
              class="chakra-text css-0"
            >
              February 10, 2025
            </span>
          </div>
          <span
            class="css-19p4nd6"
          >
            <svg
              class="chakra-icon css-1fygt0h"
              fill="none"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M10 0C8.02219 0 6.08879 0.58649 4.4443 1.6853C2.79981 2.78412 1.51809 4.3459 0.761209 6.17317C0.00433284 8.00043 -0.193701 10.0111 0.192152 11.9509C0.578004 13.8907 1.53041 15.6725 2.92894 17.0711C4.32746 18.4696 6.10929 19.422 8.0491 19.8079C9.98891 20.1937 11.9996 19.9957 13.8268 19.2388C15.6541 18.4819 17.2159 17.2002 18.3147 15.5557C19.4135 13.9112 20 11.9778 20 10C20 8.68678 19.7413 7.38642 19.2388 6.17317C18.7363 4.95991 17.9997 3.85752 17.0711 2.92893C16.1425 2.00035 15.0401 1.26375 13.8268 0.761205C12.6136 0.258658 11.3132 0 10 0ZM10 18C8.41775 18 6.87104 17.5308 5.55544 16.6518C4.23985 15.7727 3.21447 14.5233 2.60897 13.0615C2.00347 11.5997 1.84504 9.99113 2.15372 8.43928C2.4624 6.88743 3.22433 5.46197 4.34315 4.34315C5.46197 3.22433 6.88743 2.4624 8.43928 2.15372C9.99113 1.84504 11.5997 2.00346 13.0615 2.60896C14.5233 3.21447 15.7727 4.23984 16.6518 5.55544C17.5308 6.87103 18 8.41775 18 10C18 12.1217 17.1572 14.1566 15.6569 15.6569C14.1566 17.1571 12.1217 18 10 18Z"
                fill="#FCC43E"
              />
              <path
                d="M13 9H11V5C11 4.73478 10.8946 4.48043 10.7071 4.29289C10.5196 4.10536 10.2652 4 10 4C9.73478 4 9.48043 4.10536 9.29289 4.29289C9.10536 4.48043 9 4.73478 9 5V10C9 10.2652 9.10536 10.5196 9.29289 10.7071C9.48043 10.8946 9.73478 11 10 11H13C13.2652 11 13.5196 10.8946 13.7071 10.7071C13.8946 10.5196 14 10.2652 14 10C14 9.73478 13.8946 9.48043 13.7071 9.29289C13.5196 9.10536 13.2652 9 13 9Z"
                fill="#FCC43E"
              />
            </svg>
            <span
              class="chakra-text css-0"
            >
              09.00 - 11.00 AM
            </span>
          </span>
        </div>
        <div
          class="chakra-stack css-tl3ftk"
        >
          <span
            class="chakra-avatar css-1ruhx4w"
          >
            <div
              aria-label="Art"
              class="chakra-avatar__initials css-0"
              role="img"
            >
              A
            </div>
          </span>
        </div>
      </div>
    </div>
    <div
      class="chakra-card css-1i8rjqx"
    >
      <div
        class="chakra-card__header css-old1by"
      >
        <p
          class="chakra-text css-1nveigd"
        >
          Simple Past Tense
        </p>
        <span
          class="chakra-text css-bxh4bx"
        >
          English
        </span>
      </div>
      <div
        class="chakra-card__body css-19n62ks"
      >
        <div
          class="css-17dp5mp"
        >
          <div
            class="css-19p4nd6"
          >
            <svg
              class="chakra-icon css-ivbvq"
              fill="none"
              focusable="false"
              height="40"
              viewBox="0 0 40 40"
              width="40"
            >
              <path
                d="M30 8.75H27.5V7.5C27.5 7.16848 27.3683 6.85054 27.1339 6.61612C26.8995 6.3817 26.5815 6.25 26.25 6.25C25.9185 6.25 25.6005 6.3817 25.3661 6.61612C25.1317 6.85054 25 7.16848 25 7.5V8.75H21.25V7.5C21.25 7.16848 21.1183 6.85054 20.8839 6.61612C20.6495 6.3817 20.3315 6.25 20 6.25C19.6685 6.25 19.3505 6.3817 19.1161 6.61612C18.8817 6.85054 18.75 7.16848 18.75 7.5V8.75H15V7.5C15 7.16848 14.8683 6.85054 14.6339 6.61612C14.3995 6.3817 14.0815 6.25 13.75 6.25C13.4185 6.25 13.1005 6.3817 12.8661 6.61612C12.6317 6.85054 12.5 7.16848 12.5 7.5V8.75H10C9.00544 8.75 8.05161 9.14509 7.34835 9.84835C6.64509 10.5516 6.25 11.5054 6.25 12.5V30C6.25 30.9946 6.64509 31.9484 7.34835 32.6517C8.05161 33.3549 9.00544 33.75 10 33.75H30C30.9946 33.75 31.9484 33.3549 32.6517 32.6517C33.3549 31.9484 33.75 30.9946 33.75 30V12.5C33.75 11.5054 33.3549 10.5516 32.6517 9.84835C31.9484 9.14509 30.9946 8.75 30 8.75ZM8.75 12.5C8.75 12.1685 8.8817 11.8505 9.11612 11.6161C9.35054 11.3817 9.66848 11.25 10 11.25H12.5V12.5C12.5 12.8315 12.6317 13.1495 12.8661 13.3839C13.1005 13.6183 13.4185 13.75 13.75 13.75C14.0815 13.75 14.3995 13.6183 14.6339 13.3839C14.8683 13.1495 15 12.8315 15 12.5V11.25H18.75V12.5C18.75 12.8315 18.8817 13.1495 19.1161 13.3839C19.3505 13.6183 19.6685 13.75 20 13.75C20.3315 13.75 20.6495 13.6183 20.8839 13.3839C21.1183 13.1495 21.25 12.8315 21.25 12.5V11.25H25V12.5C25 12.8315 25.1317 13.1495 25.3661 13.3839C25.6005 13.6183 25.9185 13.75 26.25 13.75C26.5815 13.75 26.8995 13.6183 27.1339 13.3839C27.3683 13.1495 27.5 12.8315 27.5 12.5V11.25H30C30.3315 11.25 30.6495 11.3817 30.8839 11.6161C31.1183 11.8505 31.25 12.1685 31.25 12.5V16.25H8.75V12.5ZM31.25 30C31.25 30.3315 31.1183 30.6495 30.8839 30.8839C30.6495 31.1183 30.3315 31.25 30 31.25H10C9.66848 31.25 9.35054 31.1183 9.11612 30.8839C8.8817 30.6495 8.75 30.3315 8.75 30V18.75H31.25V30Z"
                fill="currentColor"
              />
            </svg>
            <span
              class="chakra-text css-0"
            >
              February 10, 2025
            </span>
          </div>
          <span
            class="css-19p4nd6"
          >
            <svg
              class="chakra-icon css-1fygt0h"
              fill="none"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M10 0C8.02219 0 6.08879 0.58649 4.4443 1.6853C2.79981 2.78412 1.51809 4.3459 0.761209 6.17317C0.00433284 8.00043 -0.193701 10.0111 0.192152 11.9509C0.578004 13.8907 1.53041 15.6725 2.92894 17.0711C4.32746 18.4696 6.10929 19.422 8.0491 19.8079C9.98891 20.1937 11.9996 19.9957 13.8268 19.2388C15.6541 18.4819 17.2159 17.2002 18.3147 15.5557C19.4135 13.9112 20 11.9778 20 10C20 8.68678 19.7413 7.38642 19.2388 6.17317C18.7363 4.95991 17.9997 3.85752 17.0711 2.92893C16.1425 2.00035 15.0401 1.26375 13.8268 0.761205C12.6136 0.258658 11.3132 0 10 0ZM10 18C8.41775 18 6.87104 17.5308 5.55544 16.6518C4.23985 15.7727 3.21447 14.5233 2.60897 13.0615C2.00347 11.5997 1.84504 9.99113 2.15372 8.43928C2.4624 6.88743 3.22433 5.46197 4.34315 4.34315C5.46197 3.22433 6.88743 2.4624 8.43928 2.15372C9.99113 1.84504 11.5997 2.00346 13.0615 2.60896C14.5233 3.21447 15.7727 4.23984 16.6518 5.55544C17.5308 6.87103 18 8.41775 18 10C18 12.1217 17.1572 14.1566 15.6569 15.6569C14.1566 17.1571 12.1217 18 10 18Z"
                fill="#FCC43E"
              />
              <path
                d="M13 9H11V5C11 4.73478 10.8946 4.48043 10.7071 4.29289C10.5196 4.10536 10.2652 4 10 4C9.73478 4 9.48043 4.10536 9.29289 4.29289C9.10536 4.48043 9 4.73478 9 5V10C9 10.2652 9.10536 10.5196 9.29289 10.7071C9.48043 10.8946 9.73478 11 10 11H13C13.2652 11 13.5196 10.8946 13.7071 10.7071C13.8946 10.5196 14 10.2652 14 10C14 9.73478 13.8946 9.48043 13.7071 9.29289C13.5196 9.10536 13.2652 9 13 9Z"
                fill="#FCC43E"
              />
            </svg>
            <span
              class="chakra-text css-0"
            >
              13.00 - 15.00 PM
            </span>
          </span>
        </div>
        <div
          class="chakra-stack css-tl3ftk"
        >
          <span
            class="chakra-avatar css-1ruhx4w"
          >
            <div
              aria-label="English"
              class="chakra-avatar__initials css-0"
              role="img"
            >
              E
            </div>
          </span>
        </div>
      </div>
    </div>
    <div
      class="chakra-card css-1a2sncs"
    >
      <div
        class="chakra-card__header css-old1by"
      >
        <p
          class="chakra-text css-1nveigd"
        >
          World History
        </p>
        <span
          class="chakra-text css-bxh4bx"
        >
          Class VII-B
        </span>
      </div>
      <div
        class="chakra-card__body css-19n62ks"
      >
        <div
          class="css-17dp5mp"
        >
          <div
            class="css-19p4nd6"
          >
            <svg
              class="chakra-icon css-ivbvq"
              fill="none"
              focusable="false"
              height="40"
              viewBox="0 0 40 40"
              width="40"
            >
              <path
                d="M30 8.75H27.5V7.5C27.5 7.16848 27.3683 6.85054 27.1339 6.61612C26.8995 6.3817 26.5815 6.25 26.25 6.25C25.9185 6.25 25.6005 6.3817 25.3661 6.61612C25.1317 6.85054 25 7.16848 25 7.5V8.75H21.25V7.5C21.25 7.16848 21.1183 6.85054 20.8839 6.61612C20.6495 6.3817 20.3315 6.25 20 6.25C19.6685 6.25 19.3505 6.3817 19.1161 6.61612C18.8817 6.85054 18.75 7.16848 18.75 7.5V8.75H15V7.5C15 7.16848 14.8683 6.85054 14.6339 6.61612C14.3995 6.3817 14.0815 6.25 13.75 6.25C13.4185 6.25 13.1005 6.3817 12.8661 6.61612C12.6317 6.85054 12.5 7.16848 12.5 7.5V8.75H10C9.00544 8.75 8.05161 9.14509 7.34835 9.84835C6.64509 10.5516 6.25 11.5054 6.25 12.5V30C6.25 30.9946 6.64509 31.9484 7.34835 32.6517C8.05161 33.3549 9.00544 33.75 10 33.75H30C30.9946 33.75 31.9484 33.3549 32.6517 32.6517C33.3549 31.9484 33.75 30.9946 33.75 30V12.5C33.75 11.5054 33.3549 10.5516 32.6517 9.84835C31.9484 9.14509 30.9946 8.75 30 8.75ZM8.75 12.5C8.75 12.1685 8.8817 11.8505 9.11612 11.6161C9.35054 11.3817 9.66848 11.25 10 11.25H12.5V12.5C12.5 12.8315 12.6317 13.1495 12.8661 13.3839C13.1005 13.6183 13.4185 13.75 13.75 13.75C14.0815 13.75 14.3995 13.6183 14.6339 13.3839C14.8683 13.1495 15 12.8315 15 12.5V11.25H18.75V12.5C18.75 12.8315 18.8817 13.1495 19.1161 13.3839C19.3505 13.6183 19.6685 13.75 20 13.75C20.3315 13.75 20.6495 13.6183 20.8839 13.3839C21.1183 13.1495 21.25 12.8315 21.25 12.5V11.25H25V12.5C25 12.8315 25.1317 13.1495 25.3661 13.3839C25.6005 13.6183 25.9185 13.75 26.25 13.75C26.5815 13.75 26.8995 13.6183 27.1339 13.3839C27.3683 13.1495 27.5 12.8315 27.5 12.5V11.25H30C30.3315 11.25 30.6495 11.3817 30.8839 11.6161C31.1183 11.8505 31.25 12.1685 31.25 12.5V16.25H8.75V12.5ZM31.25 30C31.25 30.3315 31.1183 30.6495 30.8839 30.8839C30.6495 31.1183 30.3315 31.25 30 31.25H10C9.66848 31.25 9.35054 31.1183 9.11612 30.8839C8.8817 30.6495 8.75 30.3315 8.75 30V18.75H31.25V30Z"
                fill="currentColor"
              />
            </svg>
            <span
              class="chakra-text css-0"
            >
              February 10, 2025
            </span>
          </div>
          <span
            class="css-19p4nd6"
          >
            <svg
              class="chakra-icon css-1fygt0h"
              fill="none"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M10 0C8.02219 0 6.08879 0.58649 4.4443 1.6853C2.79981 2.78412 1.51809 4.3459 0.761209 6.17317C0.00433284 8.00043 -0.193701 10.0111 0.192152 11.9509C0.578004 13.8907 1.53041 15.6725 2.92894 17.0711C4.32746 18.4696 6.10929 19.422 8.0491 19.8079C9.98891 20.1937 11.9996 19.9957 13.8268 19.2388C15.6541 18.4819 17.2159 17.2002 18.3147 15.5557C19.4135 13.9112 20 11.9778 20 10C20 8.68678 19.7413 7.38642 19.2388 6.17317C18.7363 4.95991 17.9997 3.85752 17.0711 2.92893C16.1425 2.00035 15.0401 1.26375 13.8268 0.761205C12.6136 0.258658 11.3132 0 10 0ZM10 18C8.41775 18 6.87104 17.5308 5.55544 16.6518C4.23985 15.7727 3.21447 14.5233 2.60897 13.0615C2.00347 11.5997 1.84504 9.99113 2.15372 8.43928C2.4624 6.88743 3.22433 5.46197 4.34315 4.34315C5.46197 3.22433 6.88743 2.4624 8.43928 2.15372C9.99113 1.84504 11.5997 2.00346 13.0615 2.60896C14.5233 3.21447 15.7727 4.23984 16.6518 5.55544C17.5308 6.87103 18 8.41775 18 10C18 12.1217 17.1572 14.1566 15.6569 15.6569C14.1566 17.1571 12.1217 18 10 18Z"
                fill="#FCC43E"
              />
              <path
                d="M13 9H11V5C11 4.73478 10.8946 4.48043 10.7071 4.29289C10.5196 4.10536 10.2652 4 10 4C9.73478 4 9.48043 4.10536 9.29289 4.29289C9.10536 4.48043 9 4.73478 9 5V10C9 10.2652 9.10536 10.5196 9.29289 10.7071C9.48043 10.8946 9.73478 11 10 11H13C13.2652 11 13.5196 10.8946 13.7071 10.7071C13.8946 10.5196 14 10.2652 14 10C14 9.73478 13.8946 9.48043 13.7071 9.29289C13.5196 9.10536 13.2652 9 13 9Z"
                fill="#FCC43E"
              />
            </svg>
            <span
              class="chakra-text css-0"
            >
              15.00 - 17.00 PM
            </span>
          </span>
        </div>
        <div
          class="chakra-stack css-tl3ftk"
        >
          <span
            class="chakra-avatar css-1ruhx4w"
          >
            <div
              aria-label="Class VII-B"
              class="chakra-avatar__initials css-0"
              role="img"
            >
              CV
            </div>
          </span>
        </div>
      </div>
    </div>
    <button
      class="chakra-button css-1uy94lb"
      disabled=""
      type="button"
    >
      View More
    </button>
  </div>
</div>
`;
